cli/agent.go

```diff
@@ -172,6 +178,10 @@ func workspaceAgent() *cobra.Command {
 				return xerrors.Errorf("add executable to $PATH: %w", err)
 			}
 
+			if err := client.PostWorkspaceAgentVersion(cmd.Context(), version); err != nil {
```

Comment:
Suggestion: We could consider discarding this API endpoint and make the version a part of the connect-to-server process. This way the version would only be updated on successful connects. I don't really see a need to be able to post an update for the version any other time during an agents lifecycle.

Reply:
I think that scenario (`UploadWorkspaceAgentKeys`) is slightly different since it's a pre-requisite for establishing the connection over WireGuard. But I won't push for this change so feel free to omit.

My motivation behind the suggestion was simply to ensure that the version tracked in the DB is consistent with "the last version of the agent that connected to the server". With the pre-post, in essence, it becomes "the last version that tried to talk to the server", but connection could've failed.

Reply:
@ammario has a PR open for metrics to be pushed to coderd. It could be done as apart of that process. Version reporting is a sort of metric?